### DISCUSSION TITLE
Time zone settings for "pvlib - pvfactors_timeseries"

### DISCUSSION THREAD

**Discussion by YutoVego** [pvlib/pvlib-python#1751] [Q&A] [answered]
Hi
Recently I started to use your contribution referencing to; https://pvlib-python.readthedocs.io/en/stable/gallery/bifacial/plot_pvfactors_fixed_tilt.html#sphx-glr-gallery-bifacial-plot-pvfactors-fixed-tilt-py

But I had problem in my timezone settings, also below the argument looks wrong since the latitude/longitude positions in Pittsburgh, US but it shows GMT+5, not GMT minus 5.

Also I tried to use in Japan timezone but it failed, would you please give me any hint how I can correctly specify the location to get the irradiance?

Thank you in advance for your any answer, response.

Best regards,

**Comment by YutoVego** [comment]
> times = pd.date_range('2021-06-21', '2021-06-22', freq='1T', tz=**'Etc/GMT+5'**)

**Reply by kandersolar** [reply]
Hi @YutoVego, there is actually no sign error in that example page.  The signs in these timezone specifiers are the opposite of what you might expect.  For more information on timezones in python, see this page in our documentation: https://pvlib-python.readthedocs.io/en/stable/user_guide/timetimezones.html

**Reply by YutoVego** [reply]
Hi @kandersolar - thank you so much for your prompt reply.
I did not get any error but the example page program which I put above at first, run as GMT+5 regardless the latitude/longitude is GMT-5, so I thought it's not wrong but I got a perfect curving graph plot, wondering why the config parameters should be like that.

Thank you for letting me know the page and I tried to specify timezone as Tokyo specifying tz='Asia/Tokyo'
> loc = location.Location(latitude=40, longitude=140, tz='Asia/Tokyo'

But I could not get the correct timeframe curve upon the Japan time axis.
![image](https://github.com/pvlib/pvlib-python/assets/133654020/604a7005-c0ee-4d46-9d0d-64c9053813b5)

I might be misunderstanding it but would you please give me some hints/advises?

**Reply by kandersolar** [reply, marked as answer]
Here is the example page, but modified for Tokyo:

```python
import pandas as pd
from pvlib import location
from pvlib.bifacial.pvfactors import pvfactors_timeseries
import matplotlib.pyplot as plt
import warnings

# supressing shapely warnings that occur on import of pvfactors
warnings.filterwarnings(action='ignore', module='pvfactors')

times = pd.date_range('2021-06-21', '2021-06-22', freq='1T', tz='Asia/Tokyo')
loc = location.Location(latitude=35.6762, longitude=139.6503, tz=times.tz)
sp = loc.get_solarposition(times)
cs = loc.get_clearsky(times)

# example array geometry
pvrow_height = 1
pvrow_width = 4
pitch = 10
gcr = pvrow_width / pitch
axis_azimuth = 180
albedo = 0.2

irrad = pvfactors_timeseries(
    solar_azimuth=sp['azimuth'],
    solar_zenith=sp['apparent_zenith'],
    surface_azimuth=180,  # south-facing array
    surface_tilt=20,
    axis_azimuth=90,  # 90 degrees off from surface_azimuth.  270 is ok too
    timestamps=times,
    dni=cs['dni'],
    dhi=cs['dhi'],
    gcr=gcr,
    pvrow_height=pvrow_height,
    pvrow_width=pvrow_width,
    albedo=albedo,
    n_pvrows=3,
    index_observed_pvrow=1
)

# turn into pandas DataFrame
irrad = pd.concat(irrad, axis=1)

irrad[['total_inc_back', 'total_abs_back']].plot()
plt.ylabel('Irradiance [W m$^{-2}$]')
```

Here is the plot I get.  It looks right to me (roughly centered on 12:00):
![image](https://github.com/pvlib/pvlib-python/assets/57452607/67271ba9-0e3a-4f70-ab8e-d3d794953297)

Could you post your code that is generating the incorrect curves/times?  I would be happy to take a look at it.

**Reply by YutoVego** [reply]
Thank you so much for showing the example code, very kind! Let me try

**Reply by YutoVego** [reply]
Now it worked well, thank you again for your kind reply. Let me close this Q&A (or from your side)
